lib/semantics/check-do.cc

```diff
@@ -219,6 +214,46 @@ class DoConcurrentBodyEnforce {
   }
 
 private:
+  void CheckAllocateObjectIsCoarray(
+      const parser::AllocateObject &allocateObject, const char *stmtType) {
```

Comment:
I think it would be better if `stmtType` were an enumeration, like with `CYCLE` and `EXIT`. It could even be the same type.

Reply:
Good idea.  I've made the associated changes.